### PR TITLE
Chore/tidy oooop

### PR DESCRIPTION
**Pull request by jjmschofield**
TLDR; https://www.youtube.com/watch?v=ssq0Nczgl-0&ab_channel=90RODan

This is a collection of updates aimed at modernizing the project and moving it towards a state where multiple people can contribute safely.

See previous PRs for more:
* [tests and local dev](https://github.com/jjmschofield/fate_character_generator/pull/1)
* [dep updates](https://github.com/jjmschofield/fate_character_generator/pull/2)
* [linting fixes and github action](https://github.com/jjmschofield/fate_character_generator/pull/3)
* [typescript and character store - plus a bunch more linting fixes, sorry](https://github.com/jjmschofield/fate_character_generator/pull/4)

### CODE REVIEW

File: .github/workflows/webpack.yml
Comment (GideonBrimleaf) [inline .github/workflows/webpack.yml:1]: This is cool - will the checks show up as part of future PRs?  I don't see them having run as part of this one...

File: src/App.tsx
Comment (GideonBrimleaf) [inline src/App.tsx:1]: Can I ask what the benefits are to just converting `App` and `index` to Typsescript rather than just moving the whole project across?

File: src/components/__snapshots__/CharacterDetail.test.js.snap
Comment (GideonBrimleaf) [inline src/components/__snapshots__/CharacterDetail.test.js.snap:1]: This is cool - not come across Jest snapshots before!

File: src/index.tsx [hunk 1]
```diff
@@ -3,10 +3,17 @@ import ReactDOM from 'react-dom';
 import './index.css';
 import App from './App';
 import * as serviceWorker from './serviceWorker';
+import { FirebaseCharacterStore } from './models/Character';
+
+const characterStore = new FirebaseCharacterStore(
+  {
+    devServer: process.env.NODE_ENV === 'production' ? null : 'http://localhost:8080/characters',
```
Comment (GideonBrimleaf) [inline src/index.tsx:10]: This is nice and neatly sidesteps the massive inconsistency I added by running an express app in Dev

**Review by GideonBrimleaf — COMMENTED** [review]
Hey @jjmschofield - thanks for all your help on this!  Just had a couple of questions but otherwise happy to approve :)

File: .github/workflows/webpack.yml
Comment (jjmschofield) [reply .github/workflows/webpack.yml:1]: yup it should do, I suspect that that gtihub actions are set to not run for a forked PR (correct thing to do - otherwise a malicious actor could use them to dump tokens by just forking and submitting PR's)

see the commit status' here https://github.com/jjmschofield/fate_character_generator/pull/4/commits for how it should look once merged

File: src/App.tsx
Comment (jjmschofield) [reply src/App.tsx:1]: just doing it a bit at a time to reduce change surface area - so the only real benefit is speed / efficiency :)  

My thinking here was to get TS into the critical areas to add safety - largely because the data model is effectively driven by JS when using Firebase (which scares me).

I'd suggest updating components as you touch them so that gradually the balance shifts

File: src/components/__snapshots__/CharacterDetail.test.js.snap
Comment (jjmschofield) [reply src/components/__snapshots__/CharacterDetail.test.js.snap:1]: they are a sweet way to get coverage for general output, I wouldn't use them for detailed logic testing - they are a handy warning that something is off, but too many and they just become _noise_ because they change too much

File: src/index.tsx (hunk 1 above)
Comment (jjmschofield) [reply src/index.tsx:10]: glad you like :) I nearly went full hog and implemented a local and firebase implementation but decided it wasn't a useful use of time

might be sweet to see what it would take to just use firebase in local dev, saves on extending a server fake over time I guess

**Comment by jjmschofield** [comment]
how's that all looking - shall we hit the merge button?
